test/test_nn.py

```diff
@@ -7699,6 +7701,39 @@ def test_adaptive_log_softmax(self):
         out = asfm.predict(x)
         self.assertEqual(out, asfm.log_prob(x).argmax(dim=1))
 
+    def test_transformer_number_match(self):
+        # Train the simple copy task.
+        V = 11
+        num_encoder_layers = 1
+        num_decoder_layers = 1
+        d_model = 16  # embedding
+        nhead = 2  # number of heads
+        dropout = 0.20
+        d_ff = 64
+        batch_size = 400
+        milliseconds = int(round(time.time() * 1000))
+        torch.manual_seed(milliseconds)
+
+        model = Transformer(V, V, d_model=d_model, nhead=nhead, num_encoder_layers=num_encoder_layers, 
+                            num_decoder_layers=num_decoder_layers, d_ff=d_ff)
+        model_opt = NoamOpt(model.encoder.src_embed.d_model, 1, 400,
+                            torch.optim.Adam(model.parameters(), lr=2.5, betas=(0.9, 0.98), eps=1e-9))
+        criterion = LabelSmoothingLoss(size=V, padding_idx=0, smoothing=0.0)
+
+        print("Training the model...")
+        for epoch in range(10):
+            model.train()
+            train_epoch(data_gen(V, batch_size, 21), model, criterion, model_opt) 
+            model.eval()
+            valid_epoch(data_gen(V, batch_size, 6), model, criterion) 
+
+        model.eval()
+
+        for _ in range(3):    
+            src = Variable(torch.randint(1, V, (10, 1)))
```

Comment:
I thought "Variable" was deprecated

Reply:
I don't see a warning for now.

Reply:
it's deprecated -- we don't warn on some ubiquitous calls because basically all pytorch code uses them.

Reply:
yea, just `src = torch.randint(1, V, (10, 1))` is sufficient

Reply:
but again, see my comments above. a full convergence test is no place for our unit test suite, it does not catch implementation bugs.